### PR TITLE
refactor __geo_interface__

### PR DESCRIPTION
**Pull request by vincentsarago**
Second try for #79 

👋 @yellowcap could you have a look at this and let me know if you have any feedback 🙏

### CODE REVIEW

File: geojson_pydantic/features.py [hunk 1]
```diff
@@ -34,6 +34,24 @@ def set_geometry(cls, v):
             return v.__geo_interface__
         return v
 
+    @property
+    def __geo_interface__(self) -> Dict:
+        """GeoJSON-like protocol for geo-spatial (GIS) vector data.
+
+        ref: https://gist.github.com/sgillies/2217756#__geo_interface
+        """
+        geo: Dict[str, Any] = {"type": self.type, "geometry": self.geometry}
```
Comment (geospatial-jeff) [inline geojson_pydantic/features.py:43]: Doesn't this need to call `.dict()` eventually?  `self.geometry` here is a pydantic geometry model not a dictionary which I think would violate geo interface spec?

Comment (geospatial-jeff) [reply geojson_pydantic/features.py:43]: Maybe should use `geom.__geo_interface__` to get the geometry instead, like how its done in feature collections.

Comment (vincentsarago) [reply geojson_pydantic/features.py:43]: oh I think it should call `geometry.__geo_interface__`

**Comment by geospatial-jeff** [comment]
I like the idea of propagating `__geo_interface__` calls up through geoms/features/geometry collections/feature collections but we should be consistent.  Only one that doesn't follow this convention is `Feature.__geo_interface__`.

**Review by geospatial-jeff — APPROVED** [review]
LGTM! 👍

**Comment by eseglem** [comment]
I would suggest consistent typing of `__geo_interface__` across the board: `def __geo_interface__(self) -> Dict[str, Any]:` 3 of the 4 are just `Dict`. Otherwise I like it.

**Pull request merged**